infrastructure/modules/lambda/main.tf

```
@@ -32,6 +33,80 @@ resource "aws_cloudwatch_log_group" "lambda_logs" {
   retention_in_days = 1
 }
 
+data "aws_caller_identity" "current" {}
+
+data "aws_iam_policy_document" "admin" {
+  statement {
+    sid    = "AllowRootAccountAccess"
+    effect = "Allow"
+
+    principals {
+      type = "AWS"
+      identifiers = [
+        "arn:aws:iam::${data.aws_caller_identity.current.account_id}:root"
+      ]
+    }
+
+    actions = [
+      "kms:*"
+    ]
+
+    resources = ["*"]
+  }
+
+  statement {
+    sid    = "AllowLambdaExecutionRole"
+    effect = "Allow"
+
+    principals {
+      type = "AWS"
+      identifiers = [
+        aws_iam_role.lambda_execution_role.arn
+      ]
+    }
+
+    actions = [
+      "kms:Encrypt",
+      "kms:Decrypt",
+      "kms:GenerateDataKey"
+    ]
+
+    resources = ["*"]
+  }
+}
+
+data "aws_iam_policy_document" "lambda" {
```

We're in the lambda module, so this should reference kms in the name e.g.:
```suggestion
data "aws_iam_policy_document" "kms_usage" {
```